traffic_ops/testing/api/v4/deliveryservices_test.go

```diff
@@ -171,13 +173,278 @@ func cleanUp(t *testing.T, ds tc.DeliveryServiceV4, oldCDNID int, newCDNID int)
+	}
+}
+
+func DeleteCDNOldSSLKeys(t *testing.T) {
+	cdn := createBlankCDN("sslkeytransfer", t)
+
+	types, _, err := TOSession.GetTypeByName("HTTP", nil)
 	if err != nil {
-		t.Error(err)
+		t.Fatal("unable to get types: " + err.Error())
+	}
+	if len(types) < 1 {
+		t.Fatal("expected at least one type")
 	}
-	_, _, err = TOSession.DeleteCDN(newCDNID)
+
+	customDS := tc.DeliveryServiceV4{}
+	customDS.Active = util.BoolPtr(true)
+	customDS.CDNID = util.IntPtr(cdn.ID)
+	customDS.DSCP = util.IntPtr(0)
+	customDS.DisplayName = util.StrPtr("displayName")
+	customDS.RoutingName = util.StrPtr("routingName")
+	customDS.GeoLimit = util.IntPtr(0)
+	customDS.GeoProvider = util.IntPtr(0)
+	customDS.IPV6RoutingEnabled = util.BoolPtr(false)
+	customDS.InitialDispersion = util.IntPtr(1)
+	customDS.LogsEnabled = util.BoolPtr(true)
+	customDS.MissLat = util.FloatPtr(0)
+	customDS.MissLong = util.FloatPtr(0)
+	customDS.MultiSiteOrigin = util.BoolPtr(false)
+	customDS.OrgServerFQDN = util.StrPtr("https://test.com")
+	customDS.Protocol = util.IntPtr(2)
+	customDS.QStringIgnore = util.IntPtr(0)
+	customDS.RangeRequestHandling = util.IntPtr(0)
+	customDS.RegionalGeoBlocking = util.BoolPtr(false)
+	customDS.TenantID = util.IntPtr(1)
+	customDS.TypeID = util.IntPtr(types[0].ID)
+	customDS.XMLID = util.StrPtr("dsID")
+	customDS.MaxRequestHeaderBytes = nil
+
+	ds, _, err := TOSession.CreateDeliveryService(customDS)
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
+	}
+	ds.CDNName = &cdn.Name
+	_, _, err = TOSession.GenerateSSLKeysForDS(*ds.XMLID, *ds.CDNName, tc.SSLKeyRequestFields{
+		BusinessUnit: util.StrPtr("BU"),
+		City:         util.StrPtr("CI"),
+		Organization: util.StrPtr("OR"),
+		HostName:     util.StrPtr("*.test.com"),
+		Country:      util.StrPtr("CO"),
+		State:        util.StrPtr("ST"),
+	})
+	if err != nil {
+		t.Fatalf("unable to generate sslkeys for DS %v: %v", *customDS.XMLID, err)
+	}
+	defer cleanUp(t, ds, cdn.ID, -1)
+
+	// Second DS creation
+	customDS2 := tc.DeliveryServiceV4{}
+	customDS2.Active = util.BoolPtr(true)
+	customDS2.CDNID = util.IntPtr(cdn.ID)
+	customDS2.DSCP = util.IntPtr(0)
+	customDS2.DisplayName = util.StrPtr("displayName2")
+	customDS2.RoutingName = util.StrPtr("routingName2")
+	customDS2.GeoLimit = util.IntPtr(0)
+	customDS2.GeoProvider = util.IntPtr(0)
+	customDS2.IPV6RoutingEnabled = util.BoolPtr(false)
+	customDS2.InitialDispersion = util.IntPtr(1)
+	customDS2.LogsEnabled = util.BoolPtr(true)
+	customDS2.MissLat = util.FloatPtr(0)
+	customDS2.MissLong = util.FloatPtr(0)
+	customDS2.MultiSiteOrigin = util.BoolPtr(false)
+	customDS2.OrgServerFQDN = util.StrPtr("https://test2.com")
+	customDS2.Protocol = util.IntPtr(2)
+	customDS2.QStringIgnore = util.IntPtr(0)
+	customDS2.RangeRequestHandling = util.IntPtr(0)
+	customDS2.RegionalGeoBlocking = util.BoolPtr(false)
+	customDS2.TenantID = util.IntPtr(1)
+	customDS2.TypeID = util.IntPtr(types[0].ID)
+	customDS2.XMLID = util.StrPtr("dsID2")
+	customDS2.MaxRequestHeaderBytes = nil
```

Comment:
maybe pull these and the ones below into a getCustomDs func or something to cut down on repeated lines?

Reply:
+1 on this